src/Microsoft.ML.Data/EntryPoints/EntryPointNode.cs

```diff
@@ -549,9 +549,20 @@ public float Cost
                 ch.Assert(weightColFieldType == typeof(string));
                 var inputWeight = inputInstance.GetType().GetField(weightColField).GetValue(inputInstance);
                 if (weight != (Optional<string>)inputWeight)
-                    ch.Warning(warning, "weight", label, inputWeight);
+                    ch.Warning(warning, "weight", weight, inputWeight);
                 else
-                    _inputBuilder.TrySetValue(weightColField, label);
+                    _inputBuilder.TrySetValue(weightColField, weight);
+            }
+            var nameColField = _inputBuilder.GetFieldNameOrNull("NameColumn");
+            if (!string.IsNullOrEmpty(name) && !string.IsNullOrEmpty(nameColField))
+            {
+                var nameColFieldType = _inputBuilder.GetFieldTypeOrNull(nameColField);
+                ch.Assert(nameColFieldType == typeof(string));
+                var inputName = inputInstance.GetType().GetField(nameColField).GetValue(inputInstance);
+                if (name != (Optional<string>)inputName)
+                    ch.Warning(warning, "name", name, inputName);
+                else
+                    _inputBuilder.TrySetValue(nameColField, weight);
```

Comment:
>weight [](start = 60, length = 6)

name!

Reply:
Thank you for finding!

---
In reply to: [199586356](https://github.com/dotnet/machinelearning/pull/467#discussion_r199586356) [](ancestors = 199586356)